### PR TITLE
add group for selection logic

### LINKED ISSUE #6825: [CT-1992] Support `group:` selection method
Groups can be selected (`group:`), similar to the `tag:` or `fqn:` selection methods.

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #6825 

### Description

Adds group as a selection method.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: .changes/unreleased/Features-20230224-134533.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Features
+body: Add abilaity to select by group resource
```
Comment (emmyoop) [inline .changes/unreleased/Features-20230224-134533.yaml:2]:
```suggestion
body: Add ability to select by group resource
```

File: core/dbt/graph/selector_methods.py [hunk 2]
```diff
@@ -143,6 +144,15 @@ def non_source_nodes(
             self.metric_nodes(included_nodes),
         )
 
+    def groupable_nodes(
+        self,
+        included_nodes: Set[UniqueId],
+    ) -> Iterator[Tuple[UniqueId, Union[ManifestNode, Metric]]]:
+        yield from chain(
+            self.parsed_nodes(included_nodes),
+            self.metric_nodes(included_nodes),
```
Comment (gshank) [inline core/dbt/graph/selector_methods.py:153]: Metric nodes don't have a "group" field.

**Review by gshank — CHANGES_REQUESTED** [review]
It looks good, except metrics don't have a group field so shouldn't be in "groupable_nodes".

File: core/dbt/graph/selector_methods.py (hunk 2 above)
Comment (MichelleArk) [reply core/dbt/graph/selector_methods.py:153]: Metric nodes should have a "group" field as per https://github.com/dbt-labs/dbt-core/pull/7069#discussion_r1119179657. Thank you @gshank for fixing that as part of: https://github.com/dbt-labs/dbt-core/pull/7069!

Comment (gshank) [reply core/dbt/graph/selector_methods.py:153]: I've added it in my pull request. I guess it doesn't have to be handled in this pr.

**Review by gshank — APPROVED** [review]

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**